include/ironside/se/update.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_INCLUDE_IRONSIDE_SE_UPDATE_H__
+#define NRF_INCLUDE_IRONSIDE_SE_UPDATE_H__
+
+/* Temporary until https://github.com/zephyrproject-rtos/zephyr/pull/88937 */
+#define IRONSIDE_SE_UPDATE_SERVICE_INVALID_PARAMS 0xf44100
+#define IRONSIDE_SE_UPDATE_SERVICE_FAILED 0xf44101
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define IRONSIDE_SE_UPDATE_MANIFEST_LENGTH  (256) /* in bytes */
+#define IRONSIDE_SE_UPDATE_PUBKEY_LENGTH    (32)  /* in bytes */
+#define IRONSIDE_SE_UPDATE_SIGNATURE_LENGTH (64)  /* in bytes */
+
+/* IRONside call identifiers with implicit versions.
+ *
+ * With the initial "version 0", the service ABI is allowed to break until the
+ * first public release of IRONside SE.
+ */
+#define IRONSIDE_CALL_ID_UPDATE_SERVICE_V0 1
+
+/**
+ * @brief Layout of update blob.
+ */
+struct ironside_se_update_blob {
+	uint8_t manifest[IRONSIDE_SE_UPDATE_MANIFEST_LENGTH];
+	uint8_t pubkey[IRONSIDE_SE_UPDATE_PUBKEY_LENGTH];
+	uint8_t signature[IRONSIDE_SE_UPDATE_SIGNATURE_LENGTH];
+	uint32_t firmware[];
+};
+
+/**
+ * @brief Request a firmware upgrade of the IRONside SE
+ *
+ * This will invoke the IRONside SE update service. The device must be restarted for the update
+ * to be installed. Check the update status in the application boot report to see if the update
+ * was successfully installed.
+ *
+ * @param update Pointer to update blob
+ *
+ * @returns -IRONSIDE_SE_UPDATE_SERVICE_INVALID_PARAMS if @ref update is invalid.
+ * @returns -IRONSIDE_SE_UPDATE_SERVICE_FAILED if writing update parameter to SICR failed.
+ * @returns 0 otherwise
```

Comment:
Should probably include the (positive) return values from IRONside call status

Reply:
Added an entry